File: java/app/src/main/java/com/cleversolutions/ads/android/sampleapp/SampleApplication.java
package com.cleversolutions.ads.android.sampleapp;

import android.app.Application;
import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;

import com.cleversolutions.ads.AdSize;
import com.cleversolutions.ads.AdTypeFlags;
import com.cleversolutions.ads.MediationManager;
import com.cleversolutions.ads.OnInitializationListener;
import com.cleversolutions.ads.android.CAS;
import com.cleversolutions.basement.CASAnalytics;

import org.jetbrains.annotations.NotNull;

public class SampleApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        // (Optional) Apply Analytics receiver
        CASAnalytics.INSTANCE.setHandler((eventName, content) -> {
            Log.i(SampleActivity.TAG, "Analytics log event " + eventName);
            for (String key : content.keySet()) {
                Log.i(SampleActivity.TAG, "\t $key:" + content.get(key));
            }
        });

        // Set Ads Settings
        CAS.getSettings().setDebugMode(true);
        CAS.getSettings().setAnalyticsCollectionEnabled(true);
        CAS.getSettings().setAllowInterstitialAdsWhenVideoCostAreLower(true);

        // Initialize SDK
        MediationManager manager = CAS.buildManager()
                .withManagerId("demo")
                .withEnabledAdTypes(AdTypeFlags.Banner | AdTypeFlags.Interstitial | AdTypeFlags.Rewarded)
                .withTestAdMode(true)
                .withInitListener((success, error) -> {
                    Log.i(SampleActivity.TAG, "CAS initialize success: " + success + " with error: " + error);
                })
                .initialize(this);

        // Set banner size
        manager.setBannerSize(AdSize.BANNER);
    }
}

File: kotlin/app/src/main/java/com/cleversolutions/ads/android/sampleapp/SampleApplication.kt
package com.cleversolutions.ads.android.sampleapp

import android.app.Application
import android.util.Log
import com.cleversolutions.ads.*
import com.cleversolutions.ads.android.CAS

class SampleApplication : Application() {

    companion object {
        const val TAG = "CAS Sample"
        lateinit var adManager: MediationManager
    }

    override fun onCreate() {
        super.onCreate()

        // Set Ads Settings
        CAS.settings.debugMode = true
        CAS.settings.taggedAudience = Audience.NOT_CHILDREN

        // Set Manual loading mode to disable auto requests
        //CAS.settings.loadingMode = LoadingManagerMode.Manual

        // Initialize SDK
        adManager = CAS.buildManager()
            .withManagerId("demo")
            .withTestAdMode(true)
            .withAdTypes(AdType.Banner, AdType.Interstitial, AdType.Rewarded)
            .withConsentFlow(
                ConsentFlow(isEnabled = true)
                //.withPrivacyPolicy("https://url")
            )
            .withCompletionListener {
                if (it.error == null)
                    Log.d(TAG, "Ad manager initialized")
                else
                    Log.d(TAG, "Ad manager initialization failed: " + it.error)
            }
            .initialize(this)
    }

}
